lib/Remotery.c

```diff
@@ -3665,9 +3675,12 @@ static rmtError ThreadSampler_Constructor(ThreadSampler* thread_sampler)
     thread_sampler->next = NULL;
 
     // Set the initial name to Thread0 etc.
-    thread_sampler->name[0] = 0;
-    strncat_s(thread_sampler->name, sizeof(thread_sampler->name), "Thread", 6);
-    itoahex_s(thread_sampler->name + 6, sizeof(thread_sampler->name) - 6, AtomicAdd(&countThreads, 1));
+    GetDebuggerThreadName(thread_sampler->name);
```

Comment:
Hmm, thinking about this, I've spotted a bug.

By default the thread name inherits the program name, truncated to 15 characters (http://man7.org/linux/man-pages/man3/pthread_setname_np.3.html).

In my code, I set the thread name when creating my threads, but others might not. Therefore all they'll see is many threads with their program name (since thread_sampler->name will never be zero length). In this case the original behaviour would be preferred.

Would it be ok if I made this feature a compile time optional feature, that is off my default?